tests/unit/File/ReadWriteHandleTest.php

```diff
@@ -90,10 +89,22 @@ public function testMustCreateExistingFile(): void
 
     public function testAppendToNonExistingFile(): void
     {
-        $this->expectException(File\Exception\NotFoundException::class);
-        $this->expectExceptionMessage('is not found.');
+        $temporary_file = Filesystem\create_temporary_file();
+        Filesystem\delete_file($temporary_file);
+
+        static::assertFalse(Filesystem\is_file($temporary_file));
 
-        new File\ReadWriteHandle(Env\temp_dir() . '/' . SecureRandom\string(20), File\WriteMode::APPEND);
+        $handle = File\open_read_write($temporary_file, File\WriteMode::APPEND);
+        $handle->tryWrite('hello');
```

Comment:
```suggestion
        $handle->writeAll('hello');
```

try write can fail without an exception